udata/tasks.py

```diff
@@ -93,7 +93,8 @@ def init_app(app):
     from udata.models import db
     with app.app_context():
         default_url = 'mongodb://{host}:{port}'.format(
-            host=db.connection.host, port=db.connection.port)
+            host=db.connection.client.address[0],
+            port=db.connection.client.address[1])
```

Comment:
May be simpler with `'mongodb://{0}:{1}'.format(*db.connection.client.address)` (only one call)